### ISSUE TITLE
Users confusing torch.distributed.nn.functional with functional collectives

### ISSUE THREAD

**Issue by ezyang** [pytorch/pytorch#119455]
### 🐛 Describe the bug

@IvanKobzarev was like "Dynamo can trace functional collectives, why isn't my code working?" And it turns out they were calling torch.distributed.nn.functional, which is not funcols.

A few things we can do:

1. Make torch.distributed.nn.functional work with PT2. Or if it's hard, make a better error message than choking on untraceable code on the inside
2. Somehow make it clear that nn.functional is not the same as functional collectives? IDK seems confusing

### Versions

main

cc @awgu @wanchaol @fegin @fduwjj @wz337 @wconstab @d4l3k @pragupta @msaroufim @dcci @aditvenk @xmfan @chauhang @penguinwu @mrshenli @pritamdamania87 @zhaojuanmao @satgera @rohan-varma @gqchen @aazzolini @osalpekar @jiayisuse @H-Huang @kwen2501 @XilunWu @tianyu-l @yf225 @bdhirsh @anijain2305 @zou3519

**Comment by wconstab** [comment]
well
We could rename funcol but its a little annoying this late in the game. We should probably do (1) regardless

**Comment by wanchaol** [comment]
torch.distributed.nn.functional was never added to our doc and I think very few users use or even aware of it. IMO we should throw deprecation messages on this set of APIs and route the existing small user groups to use our new functional collectives. 

I feel at some point a single entry of torch.distributed.functional that have all our new functional collective APIs make things more clean.

**Comment by wconstab** [comment]
Oh, I misread the issue as confusion with torch.nn.functional

Obviously that's a bad sign! In favor of depreciation

**Comment by wconstab** [comment]
no, we should actually fix this issue. lets put this on our next BE day. cc @fegin

**Comment by wconstab** [comment]
IMO this issue isn't closed, we still have the confusing 2 sets of apis.  The only thing that changed is now dynamo gives a clear error if you use the wrong one.
